src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -480,6 +479,8 @@ protected virtual string GetSuspensionPointDisplayName(SyntaxNode node, EditKind
                 var syntacticEdits = topMatch.GetTreeEdits();
                 var editMap = BuildEditMap(syntacticEdits);
 
+                var activeStatementSpanTracker = document.Project.Solution.Workspace.Services.GetRequiredService<IActiveStatementSpanTracker>();
```

Comment:
>document.Project.Solution.Workspace.Services.GetRequiredService<I [](start = 49, length = 65)

It's not desirable to access Workspace off of Solution.

Reply:
We could pass in workspace services and get the service from there. This would still add an implicit dependency on a service and that's not how EnC service is designed.

Reply:
Accessing services via solution is now restricted to debug assertions.